[Privmon] Deletion Detection and Monitoring Sources FTR Test Updates

CAWilson94 opened the pull request:
This PR introduces FTR Testing for deletion detection in the privileged user monitoring, integrations sync, feature. 

### Key Updates: 

- Coverage to test when a user is deleted, they are correctly handled during a full sync window - labels updated, privileged status set to false. This also tests the full sync detection functionality. 
- Coverage to test the case of start + complete window outside of the current batch of users, soft deleting ALL users. 
- Updated utilities functions with some common helpers and constants to create the full sync window, delete integrations user and reduce hard coded values. 
- Updated default entity source tests to include AD.

### 🐞 Bugs Found & Fixed During Test Creation

**First-run handling:**
- Previously, the fallback logic inside “detect full sync” didn’t handle the first run correctly.
- When the SO had no previous sync, it treated `fullSync === lastCompletedEvent`
- and skipped the first full-sync run.
- This has been fixed to ensure the initial run executes properly.

**The fix - Last full sync source**
* Updated lastFullSync to read only from the Saved Object (SO).
* The checks for a completed marker now live inside the deletion detection logic itself.

<!--ONMERGE {"backportTargets":["9.2"]} ONMERGE-->

CAWilson94 committed:
ad source added to monitoring sources; ad and index sync markers field added to SO test

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

CAWilson94 committed:
Deletion Detection WiP

CAWilson94 committed:
Deletion detection tests and utils

CAWilson94 committed:
Merge branch 'main' into ea-14294-intsync-del-detect-ftr

CAWilson94 committed:
remove only from test

CAWilson94 committed:
update all scheduleNow and wait methods to use consolidated rynSync method

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354543)
* Commit: fe84ed0ab68f6786c3de975faf5d13ed8500e814

### Failed CI Steps
* [FTR Configs #36](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1671-ccc7-4db0-9fce-d41c44f37f09)
* [FTR Configs #44](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1671-cccd-427f-a073-b052a8213318)
* [FTR Configs #44](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1697-e1e1-49e9-8dbd-01ae880a7e47)
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1671-ccd9-4365-84ba-ee19953c5d73)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1671-ccc7-4db0-9fce-d41c44f37f09) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354543/jobs/019a1671-ccc7-4db0-9fce-d41c44f37f09/artifacts/019a1693-cdbd-462d-83b3-c8bf55e30ce3) FTR Configs #<span></span>36 / core plugins rendering service renders "core" application without user settings
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1671-cccd-427f-a073-b052a8213318) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354543/jobs/019a1671-cccd-427f-a073-b052a8213318/artifacts/019a1697-c216-40b5-a936-cf4a2b7b9bc1) FTR Configs #<span></span>44 / Entity Analytics - Privilege Monitoring @<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Entity Privilege Monitoring APIs integrations sync deletes missing users during a full sync window
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1671-ccd9-4365-84ba-ee19953c5d73) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354543/jobs/019a1671-ccd9-4365-84ba-ee19953c5d73/artifacts/019a169d-5be7-4cc1-b962-362042ec7c67) FTR Configs #<span></span>57 / Entity Analytics - Privilege Monitoring @<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Entity Privilege Monitoring APIs integrations sync deletes missing users during a full sync window
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354543#019a1697-e1e1-49e9-8dbd-01ae880a7e47) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354543/jobs/019a1697-e1e1-49e9-8dbd-01ae880a7e47/artifacts/019a16a0-d5a6-4f8b-ac33-389c35366d32) FTR Configs #<span></span>44 / Entity Analytics - Privilege Monitoring @<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Entity Privilege Monitoring APIs integrations sync deletes missing users during a full sync window

cc @CAWilson94

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354543","number":354543,"commit":"fe84ed0ab68f6786c3de975faf5d13ed8500e814"}],"number":354543}
buildkite-pr-comment-->

CAWilson94 committed:
Update lastfullSync to get only from SO, checks for completed Marker are in deletion detection itself. Allowing fallback inside initial detect full sync did not account for the first run case where SO had no sync yet, so fullSync == lastCompletedEvent and would skip first full sync run.

CAWilson94 committed:
Tests for case where start and complete window have 0 users between them: sets all users to non privileged

privmonutils:  full sync window creation to use offsets for different sync windows

Cleanup events index for okta sync windows

CAWilson94 committed:
typo

CAWilson94 committed:
Merge branch 'main' into ea-14294-intsync-del-detect-ftr

CAWilson94 committed:
typo

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354711)
* Commit: 09e4a6a48bfc077c6b5f23b2da7e593396c0cc69

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #354677](https://buildkite.com/elastic/kibana-pull-request/builds/354677) failed 4514eb0d3da442fb30a92ee727de89f7baa98ee9
* :broken_heart: [Build #354543](https://buildkite.com/elastic/kibana-pull-request/builds/354543) failed fe84ed0ab68f6786c3de975faf5d13ed8500e814

cc @CAWilson94

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354711","number":354711,"commit":"09e4a6a48bfc077c6b5f23b2da7e593396c0cc69"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354677","number":354677,"commit":"4514eb0d3da442fb30a92ee727de89f7baa98ee9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354543","number":354543,"commit":"fe84ed0ab68f6786c3de975faf5d13ed8500e814"}],"number":354711}
buildkite-pr-comment-->

CAWilson94 committed:
Merge branch 'main' into ea-14294-intsync-del-detect-ftr

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
cc: @CAWilson94

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
cc: @CAWilson94
